src/display/connection/unix.rs

```diff
@@ -51,7 +53,7 @@ fn send_msg_packet(conn: RawFd, data: &[u8], fds: &mut Vec<Fd>) -> io::Result<()
         sendmsg_loop(conn, data, &cmsgs)
     };
 
-    if res.is_ok() {
+    if res.1.is_ok() {
         fds.clear();
```

Comment:
Hey @not-a-seagull it occurred to me that my patch has a bug here with FD sending (well... or rather "there was a bug here before that my patch did not fix"? Your call).

The above `res.1.is_ok()` clears `fds` when sending was successful. But there could also have been a partial send. Thus, I think this now needs to be `offset > 0`. Otherwise, the retry of the sending will re-send the FDs again.